include/deal.II/matrix_free/evaluation_template_factory.h

```diff
@@ -76,21 +76,24 @@ namespace internal
 
 
 
-  template <int dim, typename Number, typename VectorizedArrayType>
+  template <int dim,
+            typename Number,
+            typename VectorizedArrayType,
+            typename Number2>
   struct FEFaceEvaluationGatherFactory
   {
     static void
     evaluate(const unsigned int                                n_components,
              const EvaluationFlags::EvaluationFlags            evaluation_flag,
-             const Number *                                    src_ptr,
-             const std::vector<ArrayView<const Number>> *      sm_ptr,
+             const Number2 *                                   src_ptr,
+             const std::vector<ArrayView<const Number2>> *     sm_ptr,
```

Comment:
Shouldn't we be able to avoid `Number2` here and just use `Number`, hence implying `Number != VectorizedArrayType::value_type`? I see no reason to need the underlying scalar type of `VectorizedArrayType`, unless I am missing something.

Reply:
Like this?